### ISSUE TITLE
Axios 1.2.0 response data is unreadable

### ISSUE THREAD

**Issue by spellforce** [axios/axios#5328]
### Describe the bug

> When reponse is 200, reponse.data is garbled.

code:

`const axios = require('axios');

axios
  .request({
    transitional: {
      silentJSONParsing: true,
      forcedJSONParsing: true,
      clarifyTimeoutError: false,
    },
    timeout: 0,
    xsrfCookieName: 'XSRF-TOKEN',
    xsrfHeaderName: 'X-XSRF-TOKEN',
    maxContentLength: -1,
    maxBodyLength: -1,
    env: {
      FormData: {
        LINE_BREAK: '\r\n',
        DEFAULT_CONTENT_TYPE: 'application/octet-stream',
      },
      Blob: null,
    },
    baseURL: 'https://baidu.com/',
    method: 'get',
    url: '/',
  })
  .then(function (response) {
    console.log(response.data, 'response');
  })
  .catch(function (error) {
    console.log(error);
  });
`
![image](https://user-images.githubusercontent.com/12042870/205060898-a55b8da5-877a-49de-afda-b17e36de8a88.png)

### Axios Version

1.2.0

### Adapter Version

XHR, HTTP

### Browser

Nodejs

### Browser Version

Nodejs

### Node.js Version

16.17.1

### OS

osx 13.1

### Additional Library Versions

```bash
No need
```

### Additional context/Screenshots

```bash
![image](https://user-images.githubusercontent.com/12042870/205060898-a55b8da5-877a-49de-afda-b17e36de8a88.png)
```

**Comment by riimuru** [comment]
@spellforce You could fix the problem quickly by adding `'accept-encoding: "*"` to the headers.

**Comment by roybeitner** [comment]
I find it absolutely insane that a package this popular has such a major bug for 9 days and no one does s**t.
Need to migrate to Node v18 fetch asap.

**Comment by OpenSource03** [comment]
> I find it absolutely insane that a package this popular has such a major bug for 9 days and no one is doing s**t.
> Need to migrate to Node v18 fetch asap.

Exactly, this is really getting ridiculous... Really not sure how this is even possible...

**Comment by spellforce** [comment]
> @spellforce You could fix the problem quickly by adding `'accept-encoding: "*"` to the headers.

It works. Thanks.

**Comment by spellforce** [comment]
> I find it absolutely insane that a package this popular has such a major bug for 9 days and no one is doing s**t. Need to migrate to Node v18 fetch asap.

Completely agree

**Comment by fumieval** [comment]
I reproduced this as well. This is pointing to pentan.info

**Comment by huihuimoe** [comment]
It works. Thank you.

**Comment by Apokalypt** [comment]
If a real fix (not a hack one XD ) is provided by the library tell me please :(

**Comment by simenflatby** [comment]
> If a real fix (not a hack one XD ) is provided by the library tell me please :(

@Apokalypt You can subscribe to this issue on the top right of the page. Then you will get notified when it is closed.

**Comment by akateliev** [comment]
I've lost 2 hours debugging existing code. Amazing bug. 
btw: `'accept-encoding: "*"` fixed the issue for me.

**Comment by kodsu** [comment]
This issue will be fixed this weekend hopefully
https://github.com/axios/axios/pull/5306#issuecomment-1334055495

**Comment by julian-hoch** [comment]
Same issue. You can fix for requests with accept-encoding, but not sure how to deal with post requests - these always seem to get compressed. For now switching back to 1.1.3.

**Comment by frogburger** [comment]
This is scary for two reasons: 1 it is a key problem (granted that there is a workaround), and 2 how come this version passed the regression testing (seems to be quite obvious to replicate)?

**Comment by deepak786** [comment]
Use release 1.2.1

**Comment by WayneEllery** [comment]
Yes. This is now fixed in 1.2.1

**Comment by jasonsaayman** [comment]
Please use 1.2.1
